packages/core/apps/core-apps-server-internal/src/core_app.ts

```diff
@@ -282,5 +282,20 @@ export class CoreAppsService {
         fromRoot('node_modules/@kbn/core-apps-server-internal/assets')
       );
     });
+
+    for (const [pluginName] of uiPlugins.public) {
+      const pluginInfo = uiPlugins.internal.get(pluginName);
+      if (!pluginInfo) continue; // assuming we will never hit this case; a public entry should have internal info registered
```

Comment:
Would appreciate a sanity check on this `uiPlugins.public` => `uiPlugins.internal` assumption

Reply:
Yeah, this is correct